src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+            int executeUpdate = preparedStatement.executeUpdate();
+            if (executeUpdate < MIN_UPDATE_BOOK) {
+                throw new RuntimeException(EXPECTED_CREATE + executeUpdate);
+            }
+            ResultSet resultGeneratedKey = preparedStatement.getGeneratedKeys();
+            while (resultGeneratedKey.next()) {
+                int id = resultGeneratedKey.getObject(PARAMETER_INDEX_ONE, Integer.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_SAVE_A_BOOK_MSG + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(int id) {
+        try (PreparedStatement preparedStatement =
+                     getPreparedStatement(SELECT_FROM_BOOKS_WHERE_ID)) {
+            preparedStatement.setLong(PARAMETER_INDEX_ONE, id);
+            ResultSet executeQuery = getExecuteQuery(preparedStatement);
+            if (executeQuery.next()) {
+                Book book = buildBook(executeQuery);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_GET_A_BOOK_BY_ID_MSG + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (PreparedStatement preparedStatement = getPreparedStatement(SELECT_FROM_BOOKS)) {
+            ResultSet executeQuery = getExecuteQuery(preparedStatement);
+            while (executeQuery.next()) {
+                Book book = buildBook(executeQuery);
+                BOOK_LIST.add(book);
+            }
+            return BOOK_LIST;
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_GET_A_LIST_OF_BOOK_MSG, e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (PreparedStatement statement = getPreparedStatement(UPDATE_BOOKS_QUERY)) {
+            statement.setString(PARAMETER_INDEX_ONE, book.getTitle());
+            statement.setBigDecimal(PARAMETER_INDEX_TWO, book.getPrice());
+            statement.setInt(COLUMN_INDEX_THREE, book.getId());
+            int executeUpdate = statement.executeUpdate();
+            if (executeUpdate < MIN_UPDATE_BOOK) {
+                throw new RuntimeException(EXPECTED_CREATE + executeUpdate);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_UPDATE_BOOK + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(int id) {
+        try (PreparedStatement statement = getPreparedStatement(DELETE_FROM_BOOKS_WHERE_ID)) {
+            statement.setInt(PARAMETER_INDEX_ONE, id);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_DELETE_BOOK_BY_ID_MSG + id, e);
+        }
+        return false;
```

Comment:
The `deleteById` method always returns `false`, which might not be the intended behavior. Typically, this method should return `true` if a record was successfully deleted. Consider checking the result of `statement.executeUpdate()` to determine if the deletion was successful.